src/Node/Stream.purs

```diff
@@ -74,19 +90,55 @@ onData r cb =
       Right buf ->
         pure buf
 
+read :: forall w eff. Readable w (err :: EXCEPTION | eff) -> Eff (err :: EXCEPTION | eff) (Maybe Buffer)
+read r = do
+  v <- read' r
+  case v of
+       Nothing -> return Nothing
+       Just (BufferData b) -> return (Just b)
+       Just (StringData _) -> throw "Node.Stream.read: Stream encoding should not be set"
+       Just d              -> throw $ "Expected buffer, got got: " ++ typeof d
+
+readString :: forall w eff. Readable w (err :: EXCEPTION | eff) -> Eff (err :: EXCEPTION | eff) (Maybe String)
+readString r = do
+  v <- read' r
+  case v of
+       Just (StringData s) -> return (Just s)
+       Just d              -> throw $ "Expected buffer, got got: " ++ typeof d
```

Comment:
This should match on `Nothing` too, right? Also I think you have a typo in the error message ("got" appears twice).